src/librbd/mirror/snapshot/UnlinkPeerRequest.cc

```diff
@@ -94,12 +98,23 @@ void UnlinkPeerRequest<I>::unlink_peer() {
 
   // if there is or will be no more peers in the mirror snapshot and we have
   // a more recent mirror snapshot, remove the older one
-  if ((mirror_ns->mirror_peer_uuids.count(m_mirror_peer_uuid) == 0) ||
-      (mirror_ns->mirror_peer_uuids.size() <= 1U && m_newer_mirror_snapshots)) {
+  if ((mirror_ns->mirror_peer_uuids.size() == 0 ||
+      (mirror_ns->mirror_peer_uuids.size() == 1 &&
+       mirror_ns->mirror_peer_uuids.count(m_mirror_peer_uuid) != 0)) &&
+       m_newer_mirror_snapshots) {
+    if (m_allow_remove) {
+      m_image_ctx->image_lock.unlock_shared();
+      remove_snapshot(snap_namespace, snap_name);
+      return;
+    }
```

Comment:
Let's add a dout here:
```suggestion
    } else {
      ldout(cct, 15) << "skipping removal of snapshot: snap_id=" << m_snap_id
                     << ", mirror_peer_uuid=" << m_mirror_peer_uuid
                     << ", mirror_peer_uuids=" << mirror_ns->mirror_peer_uuids
                     << dendl;
    }
```

Reply:
Done